src/core/utils/checks/charge_neutrality.hpp

```diff
@@ -0,0 +1,33 @@
+#include <cmath>
+#include <limits>
+
+namespace Utils {
+
+template<typename ParticleRange, typename Particle>
+inline void check_charge_neutrality(ParticleRange &prange) {
```

Comment:
I think you can get that easier and with a single pass if you'd use accumulate:
```c++
template<typename ParticleRange>
void check_charge_neutrality(ParticleRange prange) {
    using Particle = typename ParticleRange::value_type;

  auto const sum_and_min = std::accumulate(prange.begin(), prange.end(), {0.0, 0.0}, 
      [](std::pair<double, double> s, const Particle &p) {
           s.first += p.p.q;
           s.second = std::min(s.second, std::abs(p.p.q));

          return s;
      });
   ...
}
```

Reply:
You could also use a `std::tuple`, then you could take the result apart into two variables using `std::tie`.  Also I think `prange` should be taken by reference to const.
```cpp
template <typename ParticleRange>
void check_charge_neutrality(ParticleRange const &prange)
{
    using Particle = typename ParticleRange::value_type;

    double sum, min;
    std::tie(sum, min) =
        std::accumulate(prange.begin(), prange.end(), std::make_tuple(0.0, 0.0),
                        [](std::tuple<double, double> s, const Particle &p) {
                            std::get<0>(s) += p.p.q;
                            std::get<1>(s) =
                                std::min(std::get<1>(s), std::abs(p.p.q));
                            return s;
                        });

    ...
}
```

Reply:
tie works with pairs, but I think it's more important to make it const. Time we get C++17.